Test Branch: New: Support for Plugins

ta264 opened the pull request:
#### Database Migration

YES (46)

#### Description

add support for plugins in lidarr.

#### Todos
- [ ] Tests
- Wiki Updates

#### Issues Fixed or Closed by this PR

* Fixes #1840 

#### Additional Information for Users wishing to test

https://wiki.servarr.com/lidarr/plugins

ta264 commented:
Comment to hope it makes hotio build...

Nonobis commented:
hello @ta264 , i am relly interested by your branch ... do you have a docker file for this branch and your deemix plugin

youegraillot commented:
Thank you for making this.
I've been able to integrate this PR  in my personnal docker image and it works flawlessly !

ta264 committed:
New: Plugin support

schwartzaher commented:
What version of the plugins can I install and use right now?

allquiet-hub commented:
Are you in need of help of any kind for this feature? What I can offer is a few years of C# knowledge as a junior developer, and I also use Lidarr extensively. Meaning if you don't need help developing the feature, I can at least offer help testing stuff.

bakerboy448 committed:
New: Improve Plugin Installation and Removal Process

Fixes restart loops
reduces github bans
improves UX with messaging for restart
improves version notes

Commit:
Improved PluginService with tree support

augustuen committed:
New: Update lidarr-plugins to support net8.0 plugins

DaRacci commented:
Is there a plan to ever merge this into the master branch?

davidnewhall commented:
This breaks Unpackerr because the API has been broken in Lidarr by changing the static values of the `protocol` member. One could wish you hadn't done that and instead made a new field.

thezak48 commented:
It is not really our concern if changes break 3rd party apps. Also as this change currently only effects nightly, it is expected for things to get broken
